### PR TITLE
Fix: Fixed crash that could occur if the Libraries folder wasn't located

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**

To prevent extra work, all changes to the Files codebase must link to an approved issue marked as `Ready to build`. Please insert the issue number following the hashtag with the issue number that this Pull Request resolves.
- Closes https://files-org.sentry.io/issues/5472102996/events/dd9fe91b32134e1eaaae5fe6a1375e0d/

**Steps used to test these changes**

Stability is a top priority for Files and all changes are required to go through testing before being merged into the repo. Please include a list of steps that you used to test this PR.

NA

### CODE REVIEW

**Review by hishitetsu — COMMENTED** [review]
I have found that if I move the original library folder to a different location and create a new library, it will be created in the original library folder that was moved, not in the new library folder that is created by this PR. So, it would be better to try to find out where the library folder is located in the first place.

**Comment by yaira2** [comment]
Is there a registry key or way to locate the path?

**Comment by hishitetsu** [comment]
There is a registry key, but it says not to use this.
image

**Comment by yaira2** [comment]
That help point me in the right direciton! I ended up using PInvoke.

File: src/Files.App/Data/Items/ShellLibraryItem.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ public sealed class ShellLibraryItem
 	{
 		public const string EXTENSION = ".library-ms";
 
-		public static readonly string LibrariesPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "Microsoft", "Windows", "Libraries");
+		public static readonly string LibrariesPath = Win32PInvoke.GetFolderFromKnownFolderGUID(new Guid("1B3EA5DC-B587-4786-B4EF-BD1DC332AEAE"));
```
Comment (0x5bfa) [inline src/Files.App/Data/Items/ShellLibraryItem.cs:12]: Please create a new class named ‘CLSID’ (I was going to do so).
It enables us to add also other than special folders such as Com Interfaces and more.

File: src/Files.App/Helpers/Win32/Win32PInvoke.Methods.cs [hunk 2]
```diff
@@ -502,5 +502,22 @@ int dwFlags
 			string pszRootPath,
 			ref SHQUERYRBINFO pSHQueryRBInfo
 		);
+
+		[DllImport("shell32.dll")]
+		static extern int SHGetKnownFolderPath(
```
Comment (0x5bfa) [inline src/Files.App/Helpers/Win32/Win32PInvoke.Methods.cs:507]: Please use CsWin32…
If we in rush, no problem, I’ll do that instead.

**Review by 0x5bfa — CHANGES_REQUESTED** [review]

File: src/Files.App/Data/Items/ShellLibraryItem.cs (hunk 1 above)
Comment (yaira2) [reply src/Files.App/Data/Items/ShellLibraryItem.cs:12]: Under Win32PInovke?

File: src/Files.App/Helpers/Win32/Win32PInvoke.Methods.cs (hunk 2 above)
Comment (yaira2) [reply src/Files.App/Helpers/Win32/Win32PInvoke.Methods.cs:507]: Sounds good.

File: src/Files.App/Data/Items/ShellLibraryItem.cs (hunk 1 above)
Comment (0x5bfa) [reply src/Files.App/Data/Items/ShellLibraryItem.cs:12]: Under Files.App.Constants

Comment (yaira2) [reply src/Files.App/Data/Items/ShellLibraryItem.cs:12]: I'll leave this for you in a follow up PR.

Comment (0x5bfa) [reply src/Files.App/Data/Items/ShellLibraryItem.cs:12]: Ok...

**Review by hishitetsu — APPROVED** [review]
Aside from the code, I have confirmed that this change will solve the issue.

**Pull request merged**